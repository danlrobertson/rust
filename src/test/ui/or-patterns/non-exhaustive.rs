#![feature(or_patterns)]
//~^ WARN the feature `or_patterns` is incomplete and may cause the compiler to crash
#![crate_type="lib"]

pub enum FirstEnum {
    Foo,
    Bar,
    Baz
}

pub fn simple_enum_test(pat: Option<FirstEnum>) -> bool {
    match pat { //~ non-exhaustive patterns: `Some(Baz)` not covered
        Some(FirstEnum::Foo | FirstEnum::Bar) => true,
        None => false,
    }
}

pub fn switch_int_test(pat: Option<usize>) -> bool {
    match pat { //~ non-exhaustive patterns: `Some(_)` not covered
        Some(1 | 2) => true,
        None => false,
    }
}

pub enum SecondEnum {
    StructLike {
        first: usize,
        second: FirstEnum
    },
    CLike
}

pub fn structlike_enum_test(pat: Option<SecondEnum>) -> bool {
    match pat { //~ non-exhaustive patterns: `Some(StructLike { .. })` not covered
        Some(SecondEnum::StructLike { first: 42, .. } | SecondEnum::CLike) => true,
        None => false,
    }
}
