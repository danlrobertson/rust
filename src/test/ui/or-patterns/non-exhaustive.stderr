warning: the feature `or_patterns` is incomplete and may cause the compiler to crash
  --> $DIR/non-exhaustive.rs:1:12
   |
LL | #![feature(or_patterns)]
   |            ^^^^^^^^^^^

error[E0004]: non-exhaustive patterns: `Some(Baz)` not covered
  --> $DIR/non-exhaustive.rs:12:11
   |
LL |     match pat {
   |           ^^^ pattern `Some(Baz)` not covered
   |
   = help: ensure that all possible cases are being handled, possibly by adding wildcards or more match arms

error[E0004]: non-exhaustive patterns: `Some(_)` not covered
  --> $DIR/non-exhaustive.rs:19:11
   |
LL |     match pat {
   |           ^^^ pattern `Some(_)` not covered
   |
   = help: ensure that all possible cases are being handled, possibly by adding wildcards or more match arms

error[E0004]: non-exhaustive patterns: `Some(StructLike { .. })` not covered
  --> $DIR/non-exhaustive.rs:34:11
   |
LL |     match pat {
   |           ^^^ pattern `Some(StructLike { .. })` not covered
   |
   = help: ensure that all possible cases are being handled, possibly by adding wildcards or more match arms

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0004`.
